pkg/ruler/ruler.go

```diff
@@ -736,3 +740,26 @@ func (r *Ruler) Rules(ctx context.Context, in *RulesRequest) (*RulesResponse, er
 
 	return &RulesResponse{Groups: groupDescs}, nil
 }
+
+// AssertMaxRuleGroupsPerUser limit has not been reached compared to the current
+// number of total rule groups in input and returns an error if so.
+func (r *Ruler) AssertMaxRuleGroupsPerUser(userID string, rg int) error {
+	limit := r.limits.RulerMaxRuleGroupsPerUser(userID)
```

Comment:
What do you think to skip checking the limit if it's `<= 0`? We're used to use `0` to disable a limit. I would also mention it in the limit CLI flag description (eg. `0 disables it`).

_Same comment applies to `AssertMaxRulesPerRuleGroupPerUser`_

Reply:
Good shout! Done.